src/Microsoft.Health.Dicom.Functions/Migration/Models/MigrationBatchCreationArguments.cs

```diff
@@ -0,0 +1,65 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using EnsureThat;
+
+namespace Microsoft.Health.Dicom.Functions.Migration.Models;
+
+/// <summary>
+/// Represents the options for creating batches for blob migration.
+/// </summary>
+public sealed class MigrationBatchCreationArguments
+{
+    /// <summary>
+    /// Gets or sets the optional inclusive maximum watermark.
+    /// </summary>
+    public long? MaxWatermark { get; }
+
+    /// <summary>
+    /// Gets or sets the number of DICOM instances processed by a single activity.
+    /// </summary>
+    public int BatchSize { get; }
+
+    /// <summary>
+    /// Gets or sets the maximum number of concurrent batches processed at a given time.
+    /// </summary>
+    public int MaxParallelBatches { get; }
+
+    /// <summary>
+    /// Gets or sets the start filter stamp
+    /// </summary>
+    public DateTime StartFilterTimeStamp { get; set; }
```

Comment:
nit: don't need `set` here and below if you have ctor

Reply:
No I need the set, Otherwise the value won't be set for the activity

Reply:
Presumably that's because you named the ctor argument differently from the property. I think for newtonsoft if you name them the same, JSON.NET should figure it out.

Reply:
oh nice catch. I totally missed that.